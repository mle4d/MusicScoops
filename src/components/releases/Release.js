import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Release({ 
  releaseCover, 
  releaseId, 
  release, 
  releaseDate = 'unknown',
  artist
}) {
  return (
    <div>
      <Link to={`/songs/${artist}/${releaseId}/${release}`}>
        <img src={releaseCover} />
        <h3>{release} - {releaseDate}</h3>
      </Link>
    </div>
  );
}

Release.propTypes = {
  releaseId: PropTypes.string.isRequired,
  release: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  releaseCover: PropTypes.string.isRequired,
  artist: PropTypes.string.isRequired
};

export default Release;
