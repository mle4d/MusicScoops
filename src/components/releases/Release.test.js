import React from 'react';
import { shallow } from 'enzyme';
import Release from './Release';

describe('Release component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Release />);
    expect(wrapper).toMatchSnapshot();
  });
});
