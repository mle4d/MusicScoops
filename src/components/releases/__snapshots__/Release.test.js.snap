// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release component renders App 1`] = `
<div>
  <Link
    to="/songs/undefined/undefined/undefined"
  >
    <img />
    <h3>
       - 
      unknown
    </h3>
  </Link>
</div>
`;
